spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/EmbeddedCassandra.scala

```diff
@@ -16,48 +16,146 @@ trait EmbeddedCassandra {
   /** Switches the Cassandra server to use the new configuration if the requested configuration is different
     * than the currently used configuration. When the configuration is switched, all the state (including data) of
     * the previously running cassandra cluster is lost.
-    * @param configTemplate name of the cassandra.yaml template resource
+    * @param configTemplates name of the cassandra.yaml template resources
     * @param forceReload if set to true, the server will be reloaded fresh even if the configuration didn't change */
-  def useCassandraConfig(configTemplate: String, forceReload: Boolean = false) {
+  def useCassandraConfig(configTemplates: Seq[String], forceReload: Boolean = false) {
     import EmbeddedCassandra._
+    if (!hosts.isEmpty && configTemplates.size > hosts.size) {
+      throw new IllegalArgumentException("Configuration templates can't be more than the number of specified hosts")
+    }
     if (sys.env.get(HostProperty).isEmpty) {
-      if (currentConfigTemplate != configTemplate || forceReload) {
-        clearCache()
-        cassandra.map(_.destroy())
-        cassandra = Some(new CassandraRunner(configTemplate))
-        currentConfigTemplate = configTemplate
+      clearCache()
+
+      for (i <- configTemplates.indices) {
```

Comment:
consider using `currentConfigTemplates.zipAll(currentConfigTemplates...)` for better clarity